### ISSUE TITLE
UB when using observers with `OnAdd`

### ISSUE THREAD

**Issue by AlfredAn** [Indra-db/Flecs-Rust#163]
```rust
use flecs_ecs::prelude::*;

#[derive(Component, Debug)]
pub struct Foo(u8);

fn main() {
    let world = World::new();

    world.observer::<flecs::OnAdd, &Foo>().each(|bar: &Foo| {
        println!("OnAdd {bar:?}");
    });

    world.entity().set(Foo(0));
}
```
This prints:
```
OnAdd Foo(224)
```
```
OnAdd Foo(160)
```
```
OnAdd Foo(48)
```
when run three separate times, which suggests that there is UB somewhere. `OnSet` and `OnRemove` seem to work fine.

**Comment by Indra-db** [comment]
Yes, that's correct, in Flecs Core, OnAdd is triggered before the value is initialized.

It seems like behavior might need to change within Flecs itself to make it safe for the Rust binding. I will update you once this problem is discussed with Sander.

**Comment by Indra-db** [comment]
completed, thanks for the issue!
